File: javascript/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mappy = {};
    for (let i = 0; i < nums.length; i++) {
        if (target - nums[i] in mappy) {
            return [mappy[target-nums[i]], i];
        } else {
            mappy[nums[i]] = i;
        }
    }
};

File: python/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        
        mappy = {}
        for i in range(len(nums)):
            diff = target - nums[i] #ver si hay este numero "diff" en el mapa. Si hay, encontramos el two sum
            if diff in mappy:
                return mappy[diff], i #retornar index del current number y de diff
            else:
                mappy[nums[i]] = i  # aca mappeamos el index con el numero en el q estamos
        
